ibmcloud/terraform/podvm-build/ansible/playbook.yml

```diff
@@ -0,0 +1,43 @@
+#
+# (C) Copyright IBM Corp. 2022.
+# SPDX-License-Identifier: Apache-2.0
+#
+---
+
+- hosts: all
+  remote_user: root
+  environment:
+    CLOUD_PROVIDER: ibmcloud
+  tasks:
+    - name: Set COS service endpoint for IBM Cloud
+      set_fact:
+        ibmcloud_cos_service_endpoint: https://s3.{{ ibmcloud_region_name }}.cloud-object-storage.appdomain.cloud
+      when: ibmcloud_api_endpoint == "https://cloud.ibm.com"
+    - name: Set COS service endpoint for IBM Cloud test
+      set_fact:
+        ibmcloud_cos_service_endpoint: https://s3.us-west.cloud-object-storage.test.appdomain.cloud
+      when: ibmcloud_api_endpoint == "https://test.cloud.ibm.com"
+    - name: Build peer pod VM image
+      shell:
+        cmd: |
+          . ~/.bashrc
```

Comment:
I am using a x86 cluster worker node to run the terraform template.
I meet follow error:
```
null_resource.ansible (local-exec): TASK [Build peer pod VM image] *************************************************
null_resource.ansible (local-exec): fatal: [10.244.64.5]: FAILED! => {"changed": true, "cmd": ". ~/.bashrc\nmake build\n", "delta": "0:00:01.530513", "end": "2022-04-29 13:46:36.114714", "msg": "non-zero return code", "rc": 2, "start": "2022-04-29 13:46:34.584201", "stderr": "/bin/sh: 13: /root/.bashrc: shopt: not found\n/bin/sh: 21: /root/.bashrc: shopt: not found\n/bin/sh: 102: /root/.bashrc: source: not found\ngo: github.com/kata-containers/kata-containers/src/runtime@v0.0.0-00010101000000-000000000000 requires\n\tgithub.com/kata-containers/govmm@v0.0.0-20210909155007-1b60b536f3c7: missing go.sum entry; to add it:\n\tgo mod download github.com/kata-containers/govmm\ngo: github.com/kata-containers/kata-containers/src/runtime@v0.0.0-00010101000000-000000000000 requires\n\tgithub.com/kata-containers/govmm@v0.0.0-20210909155007-1b60b536f3c7: missing go.sum entry; to add it:\n\tgo mod download github.com/kata-containers/govmm\ngo: github.com/kata-containers/kata-containers/src/runtime@v0.0.0-00010101000000-000000000000 requires\n\tgithub.com/kata-containers/govmm@v0.0.0-20210909155007-1b60b536f3c7: missing go.sum entry; to add it:\n\tgo mod download github.com/kata-containers/govmm\ngo: github.com/kata-containers/kata-containers/src/runtime@v0.0.0-00010101000000-000000000000 requires\n\tgithub.com/kata-containers/govmm@v0.0.0-20210909155007-1b60b536f3c7: missing go.sum entry; to add it:\n\tgo mod download github.com/kata-containers/govmm\nmake[1]: *** [Makefile:24: agent-protocol-forwarder] Error 1\nmake: *** [Makefile:67: files/usr/local/bin/agent-protocol-forwarder] Error 2", "stderr_lines": ["/bin/sh: 13: /root/.bashrc: shopt: not found", "/bin/sh: 21: /root/.bashrc: shopt: not found", "/bin/sh: 102: /root/.bashrc: source: not found", "go: github.com/kata-containers/kata-containers/src/runtime@v0.0.0-00010101000000-000000000000 requires", "\tgithub.com/kata-containers/govmm@v0.0.0-20210909155007-1b60b536f3c7: missing go.sum entry; to add it:", "\tgo mod download github.com/kata-containers/govmm", "go: github.com/kata-containers/kata-containers/src/runtime@v0.0.0-00010101000000-000000000000 requires", "\tgithub.com/kata-containers/govmm@v0.0.0-20210909155007-1b60b536f3c7: missing go.sum entry; to add it:", "\tgo mod download github.com/kata-containers/govmm", "go: github.com/kata-containers/kata-containers/src/runtime@v0.0.0-00010101000000-000000000000 requires", "\tgithub.com/kata-containers/govmm@v0.0.0-20210909155007-1b60b536f3c7: missing go.sum entry; to add it:", "\tgo mod download github.com/kata-containers/govmm", "go: github.com/kata-containers/kata-containers/src/runtime@v0.0.0-00010101000000-000000000000 requires", "\tgithub.com/kata-containers/govmm@v0.0.0-20210909155007-1b60b536f3c7: missing go.sum entry; to add it:", "\tgo mod download github.com/kata-containers/govmm", "make[1]: *** [Makefile:24: agent-protocol-forwarder] Error 1", "make: *** [Makefile:67: files/usr/local/bin/agent-protocol-forwarder] Error 2"], "stdout": "cd \"../..\" && make agent-protocol-forwarder\nmake[1]: Entering directory '/root/cloud-api-adaptor'\nCGO_ENABLED=0 go build -tags=ibmcloud -o \"agent-protocol-forwarder\" \"cmd/agent-protocol-forwarder/main.go\"\nmake[1]: Leaving directory '/root/cloud-api-adaptor'", "stdout_lines": ["cd \"../..\" && make agent-protocol-forwarder", "make[1]: Entering directory '/root/cloud-api-adaptor'", "CGO_ENABLED=0 go build -tags=ibmcloud -o \"agent-protocol-forwarder\" \"cmd/agent-protocol-forwarder/main.go\"", "make[1]: Leaving directory '/root/cloud-api-adaptor'"]}

null_resource.ansible (local-exec): PLAY RECAP *********************************************************************
null_resource.ansible (local-exec): 10.244.64.5                : ok=2    changed=0    unreachable=0    failed=1    skipped=1    rescued=0    ignored=0

╷
│ Error: local-exec provisioner error
│
│   with null_resource.ansible,
│   on main.tf line 61, in resource "null_resource" "ansible":
│   61:   provisioner "local-exec" {
│
│ Error running command 'ansible-playbook -i inventory -u root ./playbook.yml': exit status 2. Output:
│ PLAY [all] *********************************************************************
│
│ TASK [Gathering Facts] *********************************************************
│ ok: [10.244.64.5]
│
│ TASK [Set COS service endpoint for IBM Cloud] **********************************
│ ok: [10.244.64.5]
│
│ TASK [Set COS service endpoint for IBM Cloud test] *****************************
│ skipping: [10.244.64.5]
│
│ TASK [Build peer pod VM image] *************************************************
│ fatal: [10.244.64.5]: FAILED! => {"changed": true, "cmd": ". ~/.bashrc\nmake build\n", "delta": "0:00:01.530513", "end": "2022-04-29
│ 13:46:36.114714", "msg": "non-zero return code", "rc": 2, "start": "2022-04-29 13:46:34.584201", "stderr": "/bin/sh: 13: /root/.bashrc:
│ shopt: not found\n/bin/sh: 21: /root/.bashrc: shopt: not found\n/bin/sh: 102: /root/.bashrc: source: not found\ngo:
│ github.com/kata-containers/kata-containers/src/runtime@v0.0.0-00010101000000-000000000000
│ requires\n\tgithub.com/kata-containers/govmm@v0.0.0-20210909155007-1b60b536f3c7: missing go.sum entry; to add it:\n\tgo mod download
│ github.com/kata-containers/govmm\ngo: github.com/kata-containers/kata-containers/src/runtime@v0.0.0-00010101000000-000000000000
│ requires\n\tgithub.com/kata-containers/govmm@v0.0.0-20210909155007-1b60b536f3c7: missing go.sum entry; to add it:\n\tgo mod download
│ github.com/kata-containers/govmm\ngo: github.com/kata-containers/kata-containers/src/runtime@v0.0.0-00010101000000-000000000000
│ requires\n\tgithub.com/kata-containers/govmm@v0.0.0-20210909155007-1b60b536f3c7: missing go.sum entry; to add it:\n\tgo mod download
│ github.com/kata-containers/govmm\ngo: github.com/kata-containers/kata-containers/src/runtime@v0.0.0-00010101000000-000000000000
│ requires\n\tgithub.com/kata-containers/govmm@v0.0.0-20210909155007-1b60b536f3c7: missing go.sum entry; to add it:\n\tgo mod download
│ github.com/kata-containers/govmm\nmake[1]: *** [Makefile:24: agent-protocol-forwarder] Error 1\nmake: *** [Makefile:67:
│ files/usr/local/bin/agent-protocol-forwarder] Error 2", "stderr_lines": ["/bin/sh: 13: /root/.bashrc: shopt: not found", "/bin/sh: 21:
│ /root/.bashrc: shopt: not found", "/bin/sh: 102: /root/.bashrc: source: not found", "go:
│ github.com/kata-containers/kata-containers/src/runtime@v0.0.0-00010101000000-000000000000 requires",
│ "\tgithub.com/kata-containers/govmm@v0.0.0-20210909155007-1b60b536f3c7: missing go.sum entry; to add it:", "\tgo mod download
│ github.com/kata-containers/govmm", "go: github.com/kata-containers/kata-containers/src/runtime@v0.0.0-00010101000000-000000000000
│ requires", "\tgithub.com/kata-containers/govmm@v0.0.0-20210909155007-1b60b536f3c7: missing go.sum entry; to add it:", "\tgo mod download
│ github.com/kata-containers/govmm", "go: github.com/kata-containers/kata-containers/src/runtime@v0.0.0-00010101000000-000000000000
│ requires", "\tgithub.com/kata-containers/govmm@v0.0.0-20210909155007-1b60b536f3c7: missing go.sum entry; to add it:", "\tgo mod download
│ github.com/kata-containers/govmm", "go: github.com/kata-containers/kata-containers/src/runtime@v0.0.0-00010101000000-000000000000
│ requires", "\tgithub.com/kata-containers/govmm@v0.0.0-20210909155007-1b60b536f3c7: missing go.sum entry; to add it:", "\tgo mod download
│ github.com/kata-containers/govmm", "make[1]: *** [Makefile:24: agent-protocol-forwarder] Error 1", "make: *** [Makefile:67:
│ files/usr/local/bin/agent-protocol-forwarder] Error 2"], "stdout": "cd \"../..\" && make agent-protocol-forwarder\nmake[1]: Entering
│ directory '/root/cloud-api-adaptor'\nCGO_ENABLED=0 go build -tags=ibmcloud -o \"agent-protocol-forwarder\"
│ \"cmd/agent-protocol-forwarder/main.go\"\nmake[1]: Leaving directory '/root/cloud-api-adaptor'", "stdout_lines": ["cd \"../..\" && make
│ agent-protocol-forwarder", "make[1]: Entering directory '/root/cloud-api-adaptor'", "CGO_ENABLED=0 go build -tags=ibmcloud -o
│ \"agent-protocol-forwarder\" \"cmd/agent-protocol-forwarder/main.go\"", "make[1]: Leaving directory '/root/cloud-api-adaptor'"]}
│
│ PLAY RECAP *********************************************************************
│ 10.244.64.5                : ok=2    changed=0    unreachable=0    failed=1    skipped=1    rescued=0    ignored=0
│
│
```
Is there anything I missed?

Reply:
After I use my client machine, the issue is gone.
```
...
Plan: 1 to add, 0 to change, 0 to destroy.
╷
│ Warning: Value for undeclared variable
│
│ The root module does not declare a variable named "ssh_key_name" but a value was found in file "terraform.tfvars". If you meant to use this value, add a
│ "variable" block to the configuration.
│
│ To silence these warnings, use TF_VAR_... environment variables to provide certain "global" settings to all configurations in your organization. To reduce
│ the verbosity of these warnings, use the -compact-warnings option.
╵

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

null_resource.ansible: Creating...
null_resource.ansible: Provisioning with 'local-exec'...
null_resource.ansible (local-exec): Executing: ["/bin/sh" "-c" "ansible-playbook -i inventory -u root ./playbook.yml"]

null_resource.ansible (local-exec): PLAY [all] *********************************************************************

null_resource.ansible (local-exec): TASK [Gathering Facts] *********************************************************
null_resource.ansible: Still creating... [10s elapsed]
null_resource.ansible (local-exec): ok: [10.244.64.5]

null_resource.ansible (local-exec): TASK [Set COS service endpoint for IBM Cloud] **********************************
null_resource.ansible (local-exec): ok: [10.244.64.5]

null_resource.ansible (local-exec): TASK [Set COS service endpoint for IBM Cloud test] *****************************
null_resource.ansible (local-exec): skipping: [10.244.64.5]

null_resource.ansible (local-exec): TASK [Build peer pod VM image] *************************************************
null_resource.ansible: Still creating... [20s elapsed]
null_resource.ansible: Still creating... [30s elapsed]
null_resource.ansible: Still creating... [40s elapsed]
null_resource.ansible: Still creating... [50s elapsed]
null_resource.ansible: Still creating... [1m0s elapsed]
null_resource.ansible: Still creating... [1m10s elapsed]
null_resource.ansible: Still creating... [1m20s elapsed]
null_resource.ansible: Still creating... [1m30s elapsed]
null_resource.ansible: Still creating... [1m40s elapsed]
null_resource.ansible: Still creating... [1m50s elapsed]
null_resource.ansible: Still creating... [2m0s elapsed]
null_resource.ansible: Still creating... [2m10s elapsed]
null_resource.ansible: Still creating... [2m20s elapsed]
null_resource.ansible: Still creating... [2m30s elapsed]
null_resource.ansible: Still creating... [2m40s elapsed]
null_resource.ansible: Still creating... [2m50s elapsed]
null_resource.ansible: Still creating... [3m0s elapsed]
null_resource.ansible: Still creating... [3m10s elapsed]
null_resource.ansible: Still creating... [3m20s elapsed]
null_resource.ansible: Still creating... [3m30s elapsed]
null_resource.ansible: Still creating... [3m40s elapsed]
null_resource.ansible: Still creating... [3m50s elapsed]
null_resource.ansible: Still creating... [4m0s elapsed]
null_resource.ansible: Still creating... [4m10s elapsed]
null_resource.ansible: Still creating... [4m20s elapsed]
null_resource.ansible: Still creating... [4m30s elapsed]
null_resource.ansible: Still creating... [4m40s elapsed]
null_resource.ansible: Still creating... [4m50s elapsed]
null_resource.ansible: Still creating... [5m0s elapsed]
null_resource.ansible: Still creating... [5m10s elapsed]
null_resource.ansible: Still creating... [5m20s elapsed]
null_resource.ansible: Still creating... [5m30s elapsed]
null_resource.ansible: Still creating... [5m40s elapsed]
null_resource.ansible: Still creating... [5m50s elapsed]
null_resource.ansible: Still creating... [6m0s elapsed]
null_resource.ansible: Still creating... [6m10s elapsed]
null_resource.ansible (local-exec): changed: [10.244.64.5]

null_resource.ansible (local-exec): TASK [Push peer pod VM image to Cloud Object Store and verify the image] *******
null_resource.ansible: Still creating... [6m20s elapsed]
null_resource.ansible: Still creating... [6m30s elapsed]
null_resource.ansible: Still creating... [6m40s elapsed]
null_resource.ansible: Still creating... [6m50s elapsed]
null_resource.ansible: Still creating... [7m0s elapsed]
null_resource.ansible: Still creating... [7m10s elapsed]
null_resource.ansible: Still creating... [7m20s elapsed]
null_resource.ansible: Still creating... [7m30s elapsed]
null_resource.ansible: Still creating... [7m40s elapsed]
null_resource.ansible: Still creating... [7m50s elapsed]
null_resource.ansible: Still creating... [8m0s elapsed]
null_resource.ansible: Still creating... [8m10s elapsed]
null_resource.ansible: Still creating... [8m20s elapsed]
null_resource.ansible: Still creating... [8m30s elapsed]
null_resource.ansible: Still creating... [8m40s elapsed]
null_resource.ansible: Still creating... [8m50s elapsed]
null_resource.ansible: Still creating... [9m0s elapsed]
null_resource.ansible: Still creating... [9m10s elapsed]
null_resource.ansible: Still creating... [9m20s elapsed]
null_resource.ansible: Still creating... [9m30s elapsed]
null_resource.ansible: Still creating... [9m40s elapsed]
null_resource.ansible: Still creating... [9m50s elapsed]
null_resource.ansible: Still creating... [10m0s elapsed]
null_resource.ansible: Still creating... [10m10s elapsed]
null_resource.ansible: Still creating... [10m20s elapsed]
null_resource.ansible: Still creating... [10m30s elapsed]
null_resource.ansible: Still creating... [10m40s elapsed]
null_resource.ansible: Still creating... [10m50s elapsed]
null_resource.ansible: Still creating... [11m0s elapsed]
null_resource.ansible: Still creating... [11m10s elapsed]
null_resource.ansible: Still creating... [11m20s elapsed]
null_resource.ansible: Still creating... [11m30s elapsed]
null_resource.ansible: Still creating... [11m40s elapsed]
null_resource.ansible: Still creating... [11m50s elapsed]
null_resource.ansible: Still creating... [12m0s elapsed]
null_resource.ansible: Still creating... [12m10s elapsed]
null_resource.ansible: Still creating... [12m20s elapsed]
null_resource.ansible: Still creating... [12m30s elapsed]
null_resource.ansible: Still creating... [12m40s elapsed]
null_resource.ansible: Still creating... [12m50s elapsed]
null_resource.ansible: Still creating... [13m0s elapsed]
null_resource.ansible: Still creating... [13m10s elapsed]
null_resource.ansible: Still creating... [13m20s elapsed]
null_resource.ansible: Still creating... [13m30s elapsed]
null_resource.ansible: Still creating... [13m40s elapsed]
null_resource.ansible: Still creating... [13m50s elapsed]
null_resource.ansible (local-exec): changed: [10.244.64.5]

null_resource.ansible (local-exec): PLAY RECAP *********************************************************************
null_resource.ansible (local-exec): 10.244.64.5                : ok=4    changed=2    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0

null_resource.ansible: Creation complete after 13m57s [id=6382913372987208068]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

Reply:
please add a note for this one, that the pod vm terraform commands only can be run from client machine.